### ISSUE TITLE
Ignore .git-ftp-ignore

### ISSUE THREAD

**Issue by hahnjo** [git-ftp/git-ftp#176]
When creating a .git-ftp-ignore, it would also be pushed when not excluded explicitly. In my opinion, no FTP deployment needs this ignore file, so could it be ignored by default?

While at it: Necessary files for git aren't needed either. Like .gitignore and .mailmap...

**Comment by mkllnk** [comment]
Good point. I think it would be a good idea to ignore these files by default. But if we have a default, you should be able to overwrite it. The simplest I can think of would be an option like:

    git config git-ftp.ignore.gitfiles false

Any toughts?

**Comment by hahnjo** [comment]
Good idea. Will you implement it per scope?

Just cosmetic: Why 2 dots in the config name? Maybe just git-ftp.ignore-gitfiles with a dash?

**Comment by mkllnk** [comment]
That's just a matter of perspective. `git config git-ftp.ignore.gitfiles false` would result in a config file written like this:
```
[git-ftp "ignore"]
gitfiles = false
```

And your version with `git config git-ftp.ignore-gitfiles` would result in:
```
[git-ftp]
ignore-gitfiles = false
```
I thought that 'ignore' is a submodule of git-ftp. But you are right, we don't have other options in the ignore module and therefore we don't need to open another subsection in the config file. If you look at git-config`s man page, you see that they use camel case for variables. So we should go with:

    git config git-ftp.ignoreGitFiles false

resulting in:
```
[git-ftp]
ignoreGitFiles = false
```

> Will you implement it per scope?

That would be a good idea. I won't implement it in the next five weeks since I have other plans, but everybody is welcome.

**Comment by pdutourgeerling** [comment]
Why would you ignore it? The file is uploaded once and because it is a hidden file it's usually not available on the web.

**Comment by hahnjo** [comment]
Well, why would you want to have it available?
It's not needed and it's (maybe) exposing information about your directory structure (location of you git-ignored config file for example; don't know whether this is critical...)

**Comment by idhamperdameian** [comment]
+1 good idea.
